readthedocs/core/middleware.py

```diff
@@ -77,3 +79,48 @@ def __call__(self, request):
             response._csp_update = update_csp_headers[url_name]
 
         return response
+
+
+class DashboardRedirectMiddleware(MiddlewareMixin):
```

Comment:
We should write new middlewares using the new style, using MiddlewareMixin is deprecated https://docs.djangoproject.com/en/4.2/topics/http/middleware/.